src/box-client.ts

```diff
@@ -174,182 +173,643 @@ function formatResponseForBatch(response) {
+		newParams.method = 'DELETE';
+		newParams.url = getFullURL(this._baseURL, path);
+		return this._makeRequest(newParams, callback);
+	}
+
+	/**
+	 * Makes an OPTIONS call to a Box API V2 endpoint
+	 *
+	 * @param {string} path - Path to an API endpoint (e.g. /files/content)
+	 * @param {?Object} params - An optional object containing request parameters
+	 * @param {APIRequest~Callback} callback - Called with API call results, or err if call failed
+	 * @returns {void}
+	 */
+	options(path: string, params: Object | null, callback: Function) {
+		var newParams = merge({}, params || {});
+		newParams.method = 'OPTIONS';
+		newParams.url = getFullURL(this._baseURL, path);
+
+		return this._makeRequest(newParams, callback);
+	}
+
+	/**
+	 * Makes a POST call to a Box API V2 upload endpoint
+	 * @param {string} path - path to an upload API endpoint
+	 * @param {?Object} params - an optional object containing request parameters
+	 * @param {?Object} formData - multipart form data to include in the upload request {@see https://github.com/mikeal/request#multipartform-data-multipart-form-uploads}
+	 * @param {APIRequest~Callback} callback - called with API call results, or an error if the call failed
+	 * @returns {void}
+	 */
+	upload(
+		path: string,
+		params: Object | null,
+		formData: Object | null,
+		callback: Function
+	) {
+		var defaults = {
+			method: 'POST',
+		};
+		var newParams = merge(defaults, params || {});
+		newParams.url = getFullURL(this._uploadBaseURL, path);
+		newParams.formData = formData;
+		newParams.timeout = this._uploadRequestTimeoutMS;
+
+		return this._makeRequest(newParams, callback);
+	}
+
+	/**
+	 * Puts the client into batch mode, which will queue calls instead of
+	 * immediately making the API request.
+	 *
+	 * DEPRECATED: Batch API is not supported and should not be used; make calls in parallel instead.
+	 *
+	 * @returns {BoxClient} Current client object
+	 */
+	batch = util.deprecate(function (this: BoxClient) {
+		/* eslint-disable no-invalid-this */
+		this._batch = [];
+		return this;
+		/* eslint-enable no-invalid-this */
+	}, 'Batch API is not supported and should not be used; make calls in parallel instead.');
+
+	/**
+	 * Executes a batch of requests.
+	 *
+	 * DEPRECATED: Batch API is not supported and should not be used; make calls in parallel instead.
+	 *
+	 * @returns {Promise<Object>} Promise resolving to the collection of batch responses
+	 */
+	batchExec = util.deprecate(function (this: BoxClient, callback: Function) {
+		/* eslint-disable no-invalid-this */
+		if (!this._batch) {
+			return Promise.reject(
+				new Error('Must start a batch before executing')
+			).asCallback(callback);
+		}
+
+		var params = {
+			body: {
+				requests: this._batch.map((batchReq: any /* FIXME */) =>
+					formatRequestForBatch(batchReq.params)
+				),
+			},
+		};
+
+		var batch: any[] = this._batch;
+		this._batch = null;
+		return this.post('/batch', params, /* callback */ undefined)
```

Comment:
Is the addition of `/* callback */ undefined` needed?

Reply:
good catch, this is actually an optional argument